mimic/model/maas_objects.py

```diff
@@ -9,33 +9,46 @@
 import string
 from uuid import uuid4
 
-from characteristic import attributes, Attribute
+import attr
 from six import text_type
+from twisted.internet.interfaces import IReactorTime
 
-from mimic.util.helper import random_hex_generator, random_port, random_string
+from mimic.util.helper import one_of_validator, random_hex_generator, random_port, random_string
 
 METRIC_TYPE_INTEGER = 'i'
 METRIC_TYPE_NUMBER = 'n'
 METRIC_TYPE_STRING = 's'
 
 
-@attributes([Attribute('agent_id', default_value=None),
-             Attribute('alarms', default_factory=collections.OrderedDict),
-             Attribute('created_at', instance_of=int),
-             Attribute('checks', default_factory=collections.OrderedDict),
-             Attribute('id',
-                       default_factory=(lambda: u'en' + random_hex_generator(4)),
-                       instance_of=text_type),
-             Attribute('ip_addresses', default_factory=dict, instance_of=dict),
-             Attribute('label', default_value=u'', instance_of=text_type),
-             Attribute('managed', default_value=False, instance_of=bool),
-             Attribute('metadata', default_factory=dict, instance_of=dict),
-             Attribute('updated_at', instance_of=int),
-             Attribute('uri', default_value=None)])
+def _is_alarm_state(instance, attribute, value):
```

Comment:
Shouldn't this just be `_is_alarm_state = one_of_validator("OK", "WARNING", "CRITICAL", "UNKNOWN")`?

If not for some reason (perhaps some tests depend on a specific `ValueError`?), a more succinct way to express this would be `if value not in ["OK", "WARNING", "CRITICAL", "UNKNOWN"]:`.

```
    if value != 'OK' and value != 'WARNING' and value != 'CRITICAL' and value != 'UNKNOWN':
```